compose/config/interpolation.py

```diff
@@ -26,9 +29,46 @@ def process_item(name, config_dict):
     )
 
 
-def interpolate_value(name, config_key, value, section, mapping):
+def _cast_interpolated(interpolated_value, expected_type):
+
+    if expected_type == "string":
+        return None
+    elif expected_type == "number":
+        try:
+            return int(interpolated_value)
+        except ValueError:
+            return None
+    elif expected_type == "boolean":
+        if "true" == interpolated_value.lower():
+            return True
+        if "false" == interpolated_value.lower():
+            return False
+        return None
+    elif expected_type == "array":
+        if isinstance(interpolated_value, list):
+            return interpolated_value
+        return json.loads("{{\"data\":{0}}}".format(interpolated_value.replace("'", "\"")))["data"]
+
```

Comment:
This seems overly complex.  I think supporting arrays is probably the most contentious part of this feature. I think we should try to keep it as straightforward as possible.

How about:

```python
return json.loads(interpolated_value)
```

Although, I wonder if it should actually be `yaml.safe_load()` since the config is yaml format. Using yaml would support json as well.